### PR TITLE
bug #2578 fix

### PR DESCRIPTION
**Pull request by x0us**
bug #2578 fix

### CODE REVIEW

**Comment by fox1t** [comment]
I am confused by the types here. @mcollina could the serializers return objects with any propreties?

**Comment by mcollina** [comment]
> I am confused by the types here. @mcollina could the serializers return objects with any properties?

yes

**Comment by fox1t** [comment]
Then I think that these types represents only the default values that we can see here: https://github.com/fastify/fastify/blob/master/lib/logger.js#L48

I suppose that if custom types are needed for serializers, https://www.fastify.io/docs/latest/TypeScript/#example-5-specifying-logger-types should be used @x0us.

Let's wait for more info from @Ethan-Arrowood :)

**Comment by Ethan-Arrowood** [comment]
These changes look good to me. +1 on expanding the docs too!

**Comment by mcollina** [comment]
Can you please add a unit test?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by fox1t — CHANGES_REQUESTED** [review]

File: types/logger.d.ts [hunk 1]
```diff
@@ -118,21 +118,21 @@ export interface FastifyLoggerOptions<
...
-    err: (err: FastifyError) => {
+    err?: (err: FastifyError) => {
       type: string;
       message: string;
       stack: string;
       [key: string]: any;
     };
-    res: (res: RawReply) => {
+    res?: (res: RawReply) => {
       statusCode: string | number;
```
Comment (fox1t) [inline types/logger.d.ts:136]:
```suggestion
      statusCode: string | number;
      [key: string]: unknown;
```

**Review by fox1t — CHANGES_REQUESTED** [review]

**Comment by fox1t** [comment]
Can you please add a test case for additional properties?

**Comment by mcollina** [comment]
@fox1t is this ok to land?

**Review by fox1t — APPROVED** [review]
LGTM

**Pull request merged**